### ISSUE TITLE
Updating a default combination trhow webservice 

### ISSUE THREAD

**Issue by Max84** [PrestaShop/PrestaShop#12459]
Webservice hasn't the same behavior than backoffice.
When i try to create or update the field default_on for a combination throw webservice, i get an error "unable to save resource"

**To Reproduce**
1. Find a product with a default combination
2. Create a new combination throw webservice with "default_on" setted to 1
3. See error : "Unable to save resource".

Let's me explain a little bit.
In Prestashop database, you've 2 fields ps_product.id_default_combination AND ps_combination.default_on.

When you perform an update throw backoffice, on database, you change id_default_combination to the new id, set default_on to null onto all combination except for the new default.
When you perform an update throw webservice, on database, you only change default_on=1 onto the new default, but due to mysql index, you'll get a sql error => unable to save resource.

I don't know what is the Prestashop core team feeling about this:
 - If I musn't update default_on throw webservice, i suggest to transform it to a read-only field. In this case the good way is to update id_default_combination into product's method.
 - If i should be able to update it, i suggest to follow the backoffice process => change id_default_combination to the new id, set default_on to null onto all combination except for the new default.

**Additionnal information**
PrestaShop version: 1.7.5.0

**Comment by khouloudbelguith** [comment]
Hi @Max84,

I tried with this script
```

<html><head><title>CRUD Tutorial - Create example</title></head><body>
<?php


/**
 * Created by PhpStorm.
 * User: khouloud.belguith
 * Date: 06/02/19
 */
// Here we define constants /!\ You need to replace this parameters
define('DEBUG', true);
ini_set('display_errors','on');
define('PS_SHOP_PATH', 'http://shop.com/');
define('PS_WS_AUTH_KEY', 'KEY');
require_once('PSWebServiceLibrary.php');
$ws= new PrestaShopWebservice(PS_SHOP_PATH, PS_WS_AUTH_KEY, DEBUG);
$xml = $ws->get(array('url' => PS_SHOP_PATH.'/api/combinations?schema=blank'));

//On modifie les données comme souhaité
$xml->children()->children()->id_product = 30;
$xml->children()->children()->quantity = 40;
$xml->children()->children()->reference = 123;
$xml->children()->children()->price = 100;
$xml->children()->children()->minimal_quantity = 1;
$xml->children()->children()->default_on = 1;

$opt['postXml'] = $xml->asXML();
$opt['resource'] = 'combinations';
$xml = $ws->add($opt);

?>
</body></html>
```
But I did not manage to reproduce the issue with PS1.7.5.0, thanks to check & feedback.

**Comment by Max84** [comment]
Hi @khouloudbelguith 

I updated your script, you'll be able to reproduce with it.
First step, i create the combination with default_on and the second step i try to set defaut_on. You'll see the problem on this step.

`<?php


/**
 * Created by PhpStorm.
 * User: khouloud.belguith
 * Date: 06/02/19
 */
// Here we define constants /!\ You need to replace this parameters
define('DEBUG', false);
ini_set('display_errors','on');
define('PS_SHOP_PATH', 'http://shop.com/');
define('PS_WS_AUTH_KEY', 'KEY');
require_once('PSWebServiceLibrary.php');
$ws= new PrestaShopWebservice(PS_SHOP_PATH, PS_WS_AUTH_KEY, DEBUG);
$xml = $ws->get(array('url' => PS_SHOP_PATH.'/api/combinations?schema=blank'));

//On modifie les données comme souhaité
$xml->children()->children()->id_product = 30;
$xml->children()->children()->quantity = 40;
$xml->children()->children()->reference = 123;
$xml->children()->children()->price = 100;
$xml->children()->children()->minimal_quantity = 1;
$xml->children()->children()->default_on = 0;

$opt['postXml'] = $xml->asXML();
$opt['resource'] = 'combinations';
$xml = $ws->add($opt);

//Get the last created id
$id = (int)$xml->combination->id;

//update the same combination
$xml = $ws->get(array('url' => PS_SHOP_PATH.'/api/combinations/' . $id));
$xml->children()->children()->default_on = 1;
$opt2['putXml'] = $xml->asXML();
$opt2['id'] = $id;
$opt2['resource'] = 'combinations';
$xml = $ws->edit($opt2);`

Thanks.

**Comment by khouloudbelguith** [comment]
Hi @Max84,

Thanks for your feedback.
I tried with the script updated but I get an error:

`Fatal error: Uncaught PrestaShopWebserviceException: This call to PrestaShop Web Services failed and returned an HTTP status of 500. That means Internal Server Error. in /projet/1750/PSWebServiceLibrary.php on line 87`

I checked in the BO, I found that this combination is created.
Thanks!

**Comment by Max84** [comment]
Yes, of course you'll get an error, due to the Prestashop "unable to save resource" and HTTP 500 return.
If you remove this line "$xml->children()->children()->default_on = 1;", you don't have anymore this error.

It's not a PSWebServiceLibrary.php bug, but an exception due to Prestashop webservice.

Do you get it ?

**Comment by khouloudbelguith** [comment]
@Max84, thanks for this clarifications.
I’ll add this to the debug roadmap so that it’s fixed. If you have already fixed it on your end or if you think you can do it, please do send us a pull request!
Thanks!
